crates/autopilot/src/run.rs

```diff
@@ -363,11 +363,27 @@ pub async fn run(args: Arguments) {
         infra::persistence::Persistence::new(args.s3.into().unwrap(), Arc::new(db.clone())).await;
     let settlement_observer =
         crate::domain::settlement::Observer::new(eth.clone(), persistence.clone());
+    let settlement_contract_start_index =
+        if let Some(DeploymentInformation::BlockNumber(settlement_contract_start_index)) =
+            eth.contracts().settlement().deployment_information()
+        {
+            settlement_contract_start_index
+        } else {
+            // If the deployment information can't be found, start from 0 (default
+            // behaviour). For real contracts, the deployment information is specified
+            // for all the networks, but it isn't specified for the e2e tests which deploy
+            // the contracts from scratch
+            0
```

Comment:
you might want to log a warning in this case (so that we would see it on startup)